cgi/CRMS.pm

```diff
@@ -165,6 +151,27 @@ sub SetupUser
   return $crms_user;
 }
 
+# Extract potentially multiple values of ENV{email} as an array ref.
+# We have seen multiple (duplicate) values of email separated by semicolons
+# coming from Shib.
+# Values are downcased, unique, nonempty strings with any "@umich.edu" stripped.
+sub extract_env_email {
```

Comment:
This is a highly optional bit of feedback...

If the string of email(s) is passed in as an argument that defaults to `$ENV{email}` you can simplify the tests (in that you don't have to reset `$ENV{email}` between tests).

```
sub extract_env_email {
my $self = shift;
my $email_str = shift || $ENV{email};
...
# in CRMS.t: 
# $ENV{email} = 'someone@umich.edu'; # no longer needed
is_deeply($crms->extract_env_email('someone@umich.edu'), ['someone'], 'strips @umich.edu');
```